Healthy/Classes/Utilities/Animatable.swift

```diff
@@ -0,0 +1,69 @@
+import UIKit
+
+/// A public protocol that contain two functions that will be affectively on button loading appearance
+/// Conform to AnyObject because this protocol will work only with classes such ex: UIButton, UIView....etc
+protocol Animatable: AnyObject {
+    
+    /// Start animating while waiting another task is done
+    /// - Parameter color: indicator color
+    func startAnimation(withColor color: UIColor)
+    
+    /// Stop animating when the task is finished
+    func stopAnimation()
+}
+
+/// Default implementation to UIButton loading indicator
+extension Animatable where Self: UIButton {
+    
+    ///A  computed property that returns the first subview that is an instance of UIActivityIndicatorView, if one exists.
+    private var activityIndicator: UIActivityIndicatorView? {
+        return subviews.first(where: { $0 is UIActivityIndicatorView }) as? UIActivityIndicatorView
+    }
+    
+    /// Start animating while waiting another task is done
+    /// We use an associated object to store the original button title when startLoading() is called.
+    /// - Parameter color: indicator color
+    func startAnimation(withColor color: UIColor = .black) {
+        let originalTitle = title(for: .normal)
+        isEnabled = false
+        setTitle("", for: .normal)
```

Comment:
We lost the title, forever 😅

Reply:
I think we can access titleLabel and update it's alpha instead?!

Reply:
> We lost the title, forever 😅

I save it to original title with associated object 
But yes I want to discuss with you a way to save our title that more efficient than objc dynamic mechanism
This solution is working fine but I think there another way to enhance

Reply:
> I think we can access titleLabel and update it's alpha instead?!

That's a great solution and make more sense.
Thanks ya ahmed ❤️🚀